documentation/library_guide/kernel_templates_main.rst

```diff
@@ -0,0 +1,17 @@
+Kernel Templates API
+####################
+
+Kernel templates is an experimental API that includes algorithms that can be configured based on various parameters, such as the number of elements to be processed by a work-item and the size of a work-group.
```

Comment:
Do we want some explicit mention that beyond just being configurable, kernel templates include algorithms which may rely on characteristics of specific devices (unlike the main oneDPL parallel algorithms)?

You refer to it partially in the next line, but it may be worthwhile to make a larger point of it.  The first KT is this esimd radix sort which relies on the presence of ESIMD (and therefore specific intel GPUs).  The main assumption for the other active KT PR single-pass scan is forward progress guarantees.

Reply:
It can be mentioned on the pages of the KT backends or individual algorithms as a part of system requirements or other notes.

I do not know which value it may bring if it is mentioned here.

Reply:
I suppose I see this "device characteristics assumption" this as at least one of the main purposes / drivers behind kernel templates as a whole in contrast to the rest of oneDPL which attempts to be quite general and work on any hardware as much as is possible.  Both this and the configurability are in pursuit of best performance.  Since this section introduces the high level topic, that is why I see this as interesting to mention here. 

I wont insist though.

Really this is a question for @akukanov, as I think it indicates how we want to position kernel templates conceptually.

Reply:
To be clear, I don't think specifics are needed here, but rather some high level statement like 

"These algorithms may rely upon assumptions which are not satisfied by all devices, and are intended to provide best performance at the expense of some generality."

Reply:
I have added this sentence with a slight change to improve cohesion.